pandas/core/internals/construction.py

```diff
@@ -159,7 +159,29 @@ def init_ndarray(values, index, columns, dtype=None, copy=False):
     # on the entire block; this is to convert if we have datetimelike's
     # embedded in an object type
     if dtype is None and is_object_dtype(values):
-        values = maybe_infer_to_datetimelike(values)
+
+        if values.ndim == 2 and values.shape[0] != 1:
+            # kludge to transpose and separate blocks
+            # unnecessary if we ever allow 2D DatetimeArray
+
+            dvals_list = [maybe_infer_to_datetimelike(values[n, :])
+                          for n in range(len(values))]
```

Comment:
make this a list comprehension & use enumerate if you must., this is very hard to read.